File: NativeAPI/Python/b2_finish_large_file.py
import requests
import json

api_url = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
file_id = ""  # Provided by b2_start_large_file
part_sha1_array = ["<sha1_of_part_1>", "<sha1_of_part_2>", "<sha1_of_part_3>"]  # See b2_upload_part

response = requests.post(
    f"{api_url}/b2api/v4/b2_finish_large_file",
    headers={"Authorization": account_authorization_token},
    json={
        "fileId": file_id,
        "partSha1Array": part_sha1_array
    }
)
response_data = response.json()
print(json.dumps(response_data, indent=2))

File: NativeAPI/Java/B2FinishLargeFile.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2FinishLargeFile {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String fileId = ""; // Provided by b2_start_large_file
        String partSha1Array = "[\"sha1_of_part_1\", \"sha1_of_part_2\", \"sha1_of_part_3\"]"; // See b2_upload_part

        String postParams = "{" +
                "\"fileId\":\"" + fileId + "\"" +
                ",\"partSha1Array\":" + partSha1Array +
                "}";

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_finish_large_file"))
                    .header("Authorization", accountAuthorizationToken)
                    .POST(HttpRequest.BodyPublishers.ofString(postParams))
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String jsonResponse = response.body();
            System.out.println(jsonResponse);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
